ui/src/tempVars/utils/replace.ts

```diff
@@ -110,26 +110,58 @@ const renderTemplate = (query: string, template: Template): string => {
   }
 }
 
+const REGEX_COMPARATORS = ['=~', '!~']
+
 const replaceAllRegex = (
   query: string,
   search: string,
   replacement: string
 ) => {
-  // check for presence of anything between two forward slashes /[your stuff here]/
-  const matches = query.match(/\/([^\/]*)\//gm)
+  let result = query
+  let i = 0
 
-  if (!matches) {
-    return query
-  }
+  while (i < result.length - 1) {
+    const chars = result[i] + result[i + 1]
+    const isStartOfRegex = REGEX_COMPARATORS.includes(chars)
 
-  return matches.reduce((acc, m) => {
-    if (m.includes(search)) {
-      const replaced = m.replace(search, replacement)
-      return acc.split(m).join(replaced)
+    if (!isStartOfRegex) {
+      i += 1
+      continue
     }
 
-    return acc
-  }, query)
+    const regexStart = findRegexStart(result, i)
+    const regexEnd = findRegexEnd(result, regexStart)
+    const regexContent = result.slice(regexStart + 1, regexEnd)
+    const replacedRegexContent = regexContent.replace(search, replacement)
+
+    result = replaceSubstring(
+      result,
+      regexStart + 1,
+      regexEnd - 1,
+      replacedRegexContent
+    )
+
+    i = findRegexEnd(result, regexStart)
+  }
+
+  return result
+}
+
+const findRegexStart = (query: string, startIndex: number): number => {
+  return startIndex + query.slice(startIndex, query.length).indexOf('/')
+}
+
+const findRegexEnd = (query: string, startIndex: number): number => {
+  return startIndex + 1 + query.slice(startIndex + 1, query.length).indexOf('/')
+}
+
+const replaceSubstring = (s: string, i: number, j: number, content: string) => {
+  // Replace everything in `s` from indices `i` up to but not including `j` with `content`
+  const split = s.split('')
+
+  split.splice(i, j - i + 1, ...content.split(''))
```

Comment:
do you not like lodash?  especially for array operations, I think it's so much easier to reason with and makes one hate JS less?
```firstBit= _.join(_.slice(split, 0, i))```
```lastBit = _.join(_.slice(split, j+1))```
```return '${firstBit}${content}${lastBit}'```

I still hated JS, but maybe tiny bit better?

Reply:
seems much clearer, will change